marimo-tutorials/Data-Science/Exploratory-Data-Analysis/Goodreads-EDA.py

```diff
@@ -0,0 +1,1271 @@
+    return
+
+
+@app.cell(hide_code=True)
+def __(df1, df2, plt, sns):
+    # Bivariate Analysis
+
+    # 1. Rating vs. Publication Year
+    # Merging dataframes assuming 'book_id' is the common column
+
+    merged_data = df2.merge(df1[["rating", "book_id"]], how="left", on="book_id")
+    sns.jointplot(x="original_publication_year", y="rating", data=merged_data)
+    plt.xlabel("Publication Year")
+    plt.ylabel("Rating")
+    plt.title("Rating vs. Publication Year in Goodreads Dataset")
+    plt.gca()
+    return (merged_data,)
+
+
+@app.cell(hide_code=True)
+def __(mo):
+    mo.md(
+        r"""
+        ```python
+        # 2. Rating vs. Number of Ratings per Book
+        # Calculate average rating per book
+        average_ratings = df1.groupby('book_id')['rating'].mean()
+        _merged_data = df2.merge(average_ratings.reset_index(), how='left', on='book_id')
+        _merged_data.rename(columns={'rating_x': 'rating'}, inplace=True)  # Avoid name conflicts
+
+        sns.jointplot(x='average_rating', y='rating', data=_merged_data)
+        plt.xlabel("Average Rating per Book")
+        plt.ylabel("Individual User Rating")
+        plt.title("Rating vs. Average Rating per Book in Goodreads Dataset")
+        plt.gca()
+        ```
+        """
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def __(df1, df2, plt, sns):
+    # 2. Rating vs. Number of Ratings per Book
+    # Calculate average rating per book
+    average_ratings = df1.groupby("book_id")["rating"].mean()
+    _merged_data = df2.merge(average_ratings.reset_index(), how="left", on="book_id")
+    _merged_data.rename(
+        columns={"rating_x": "rating"}, inplace=True
+    )  # Avoid name conflicts
+
+    sns.jointplot(x="average_rating", y="rating", data=_merged_data)
+    plt.xlabel("Average Rating per Book")
+    plt.ylabel("Individual User Rating")
+    plt.title("Rating vs. Average Rating per Book in Goodreads Dataset")
+    plt.gca()
+    return (average_ratings,)
+
+
+@app.cell
+def __():
+    # 10.	Check for Missing Value and Duplicated Rows. If required correct it.
+    # Done in the preprocessing of data.
+    return
+
+
+@app.cell(hide_code=True)
+def __(mo):
+    mo.md(r"""### 11.	How is the rating for all books distributed?""")
+    return
+
+
+@app.cell(hide_code=True)
+def __(mo):
+    mo.md(
+        r"""
+        ```python
+        # 11. How is the rating for all books distributed?
+        plt.figure(figsize=(10, 6))
+        sns.histplot(df1['rating'], bins=10, kde=True)
+        plt.title('Distribution of Ratings')
+        plt.xlabel('Rating')
+        plt.ylabel('Count')
+        plt.gca()
+        ```
+        """
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def __(df1, plt, sns):
+    # 11. How is the rating for all books distributed?
+    plt.figure(figsize=(10, 6))
+    sns.histplot(df1["rating"], bins=10, kde=True)
+    plt.title("Distribution of Ratings")
+    plt.xlabel("Rating")
+    plt.ylabel("Count")
+    plt.gca()
+    return
+
```

Comment:
I think a barchart here is better, since the available values are simply 1, 2, 3, 4, 5